meeteval/der/md_eval.py

```diff
@@ -0,0 +1,175 @@
+            logging.info(f'md-eval-22.pl not found. Trying to download it from {url}.')
+            import urllib.request
+            urllib.request.urlretrieve(url, md_eval_22)
+            logging.info(f'Wrote {md_eval_22}')
+
+    def get_details(r, h, key, tmpdir):
+        r_file = tmpdir / f'{key}.ref.rttm'
+        h_file = tmpdir / f'{key}.hyp.rttm'
+        r.dump(r_file)
+        h.dump(h_file)
+
+        cmd = [
+            'perl', f'{md_eval_22}',
+            '-c', f'{collar}',
+            '-r', f'{r_file}',
+            '-s', f'{h_file}',
+        ]
+
+        cp = subprocess.run(cmd, stdout=subprocess.PIPE,
+                            check=True, universal_newlines=True)
+        # SCORED SPEAKER TIME =4309.340250 secs
+        # MISSED SPEAKER TIME =4309.340250 secs
+        # FALARM SPEAKER TIME =0.000000 secs
+        # SPEAKER ERROR TIME =0.000000 secs
+        #  OVERALL SPEAKER DIARIZATION ERROR = 100.00 percent of scored speaker time  `(ALL)
+
+        error_rate, = re.findall(r'OVERALL SPEAKER DIARIZATION ERROR = ([\d.]+) percent of scored speaker time', cp.stdout)
+        length, = re.findall(r'SCORED SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        deletions, = re.findall(r'MISSED SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        insertions, = re.findall(r'FALARM SPEAKER TIME =([\d.]+) secs', cp.stdout)
+        substitutions, = re.findall(r'SPEAKER ERROR TIME =([\d.]+) secs', cp.stdout)
+
+        def convert(string):
+            return decimal.Decimal(string)
+
+        details = dict(
+            scored_speaker_time=convert(length),
+            # errors=float(errors),
+            missed_speaker_time=convert(deletions),
+            falarm_speaker_time=convert(insertions),
+            speaker_error_time=convert(substitutions),
+            error_rate=convert(error_rate) / 100,
+        )
+        return DiaErrorRate(**details)
+
+    per_reco = {}
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmpdir = Path(tmpdir)
+        for key in keys:
+            per_reco[key] = get_details(r[key], h[key], key, tmpdir)
+
+        md_eval = get_details(
+            meeteval.io.RTTM([line for key in keys for line in r[key]]),
+            meeteval.io.RTTM([line for key in keys for line in h[key]]),
+            '',
+            tmpdir,
+        )
+        summary = sum(per_reco.values())
+        error_rate = summary.error_rate.quantize(md_eval.error_rate)
+        if error_rate != md_eval.error_rate:
+            raise RuntimeError(
+                f'The error rate of md-eval-22.pl on all recordings '
+                f'({summary.error_rate})\n'
+                f'does not match the average error rate of md-eval-22.pl '
+                f'applied to each recording ({md_eval.error_rate}).'
+            )
+
+    from meeteval.wer.__main__ import _save_results
+    _save_results(per_reco, hypothesis_paths, per_reco_out, average_out)
+
+
+if __name__ == '__main__':
```

Comment:
Why is this necessary? Isn't `meeteval.der.__main__` enough for CLI?

Reply:
That was my old code. removed